File: PINGServer.cpp
#include <cstdlib>
#include <iostream>
#include <netinet/in.h>
#include <sys/socket.h>
#include <string.h>

#define MAXLINE 1024

void error(const char *);
void print_packet(std::string &);


int main(int argc, char **argv)
{
	if(argc != 3)
		error("incorrect number of arguments");
	int sockfd;

	unsigned short serv_port = (unsigned short)std::stoi(argv[1]);
	int loss = std::stoi(argv[2]);

	struct sockaddr_in serv_addr, cli_addr;
        char buffer[MAXLINE];
	if((sockfd = socket(AF_INET, SOCK_DGRAM, 0))<0)
		error("failed to create socket");

	memset(&serv_addr, 0, sizeof(serv_addr));
	memset(&cli_addr, 0, sizeof(cli_addr));

	serv_addr.sin_family = AF_INET;
	serv_addr.sin_addr.s_addr = INADDR_ANY;
	serv_addr.sin_port = htons(8080);

	if(bind(sockfd, (const struct sockaddr *)&serv_addr, sizeof(serv_addr)) < 0)
		error("bind failed");

	socklen_t len;
	int n;

	len = sizeof(cli_addr);

	n = recvfrom(sockfd,(char*)buffer,MAXLINE, MSG_WAITALL, (struct sockaddr *)&cli_addr, &len);
	buffer[n] = '\0';
	std::string b(buffer);
	std::cout << b << " :END\n";
	//sendto(sockfd, (const char*)hello, strlen(hello));
	std::cout  << "hello message sent\n";

	return 0;
}



// print error message and exit program
void error(const char *err_msg)
{
	std::cout << "ERROR: " << err_msg << "\n";
	exit(1);
}

File: PINGServer.py
from socket import *
import sys

def print_packet(pkt):
    info = pkt.split("|")
    print("-----------Ping Request Packet Header------------")
    print("Version:", info[0])
    print("ClientID:", info[1])
    print("SequenceNo:",info[2])
    print("Timestamp:",info[3])
    print("Size:",info[4])
    print("-----------Ping Request Packet Header-----------")
    print("Host:",info[5])
    print("Class-name:",info[6])
    print("User-name:",info[7])
    print("Rest:",info[8])
    print("-------------------------------------------")

def packet_capitalize(pkt):
    info = pkt.split('|')
    for i in range(5,8):
        info[i] = info[i].capitalize()
    return '|'.join(info)
    

def main():
    args = sys.argv

    if len(args) != 3:
        print("Not right number of args")
        exit(1)
    port = int(args[1])
    serverSocket = socket(AF_INET, SOCK_DGRAM)
    serverSocket.bind(('',port))


    while 1:
        pkt, cliAddr = serverSocket.recvfrom(2048)
        pkt = pkt.decode()
        print_packet(pkt)
        serverSocket.sendto(pkt.upper().encode(),cliAddr)
        print ("ack sent")




main()
